core/src/main/scala/better/files/Cmds.scala

```diff
@@ -124,4 +124,13 @@ object Cmds {
     } output.add(file, name.toString)
     destination
   }
+
+  def zip(files: (File, String)*)(destination: File, compressionLevel: Int)(implicit codec: Codec, d: DummyImplicit): File = {
+    for {
+      output <- new ZipOutputStream(destination.newOutputStream, codec).withCompressionLevel(compressionLevel).autoClosed
+      input <- files
+      file <- input._1.walk()
+    } output.add(file, input._2)
```

Comment:
Je comprend pas bien la question, mais il n'est pas possible de faire un `name <- input._2.walk()` car `walk` est une fonction de la class File.
Source: 
```
/**
    * Walk the directory tree recursively upto maxDepth
    *
    * @param maxDepth
    * @return List of children in BFS maxDepth level deep (includes self since self is at depth = 0)
    */
  def walk(maxDepth: Int = Int.MaxValue)(implicit visitOptions: File.VisitOptions = File.VisitOptions.default): Files =
    Files.walk(path, maxDepth, visitOptions: _*) //TODO: that ignores I/O errors?
```